src/test/java/com/kodilla/ecommercee/domainTests/ProductTestSuite.java

```diff
@@ -0,0 +1,162 @@
+    //Then
+        assertNotNull(savedGroup);
+        assertEquals("Test Product", savedProduct.getName());
+        assertEquals("Test Description", savedProduct.getDescription());
+        assertEquals(BigDecimal.valueOf(10.00), savedProduct.getPrice());
+        assertNotNull(savedProduct.getGroup());
+        assertEquals(1, savedProduct.getCarts().size());
+        assertEquals(1, savedProduct.getOrders().size());
+        assertTrue(savedProduct.isActive());
+    //CleanUp
+        productRepository.deleteAll();
+        groupRepository.deleteAll();
+    // Verify cleanup
+        assertEquals(0, productRepository.findAll().size());
+        assertEquals(0, groupRepository.findAll().size());}
+    @Test
+    public void testFindProductById() {
+    // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+    // When
+        Long productId = savedProduct.getId();
+        Product retrievedProduct = productRepository.findById(productId).orElse(null);
+
+    // Then
+        assertNotNull(retrievedProduct);
+        assertEquals(productId, retrievedProduct.getId());
+        assertEquals("Test Product", retrievedProduct.getName());
+        assertEquals("Test Description", retrievedProduct.getDescription());
+        assertEquals(BigDecimal.valueOf(10.00), retrievedProduct.getPrice());
+        assertTrue(retrievedProduct.isActive());
+    }
+    @Test
+    public void testDeleteProduct() {
+        // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+        // When
+        productRepository.delete(savedProduct);
+
+        // Then
+        Optional<Product> deletedProductOptional = productRepository.findById(savedProduct.getId());
+        Assertions.assertTrue(deletedProductOptional.isPresent());
+        Assertions.assertFalse(deletedProductOptional.get().isActive());
+    }
+    @Test
+    public void testUpdateProduct() {
+    // Given
+        Product product = new Product();
+        product.setName("Test Product");
+        product.setDescription("Test Description");
+        product.setPrice(BigDecimal.valueOf(10.00));
+        product.setActive(true);
+        Product savedProduct = productRepository.save(product);
+
+    // When
+        savedProduct.setName("Updated Product Name");
+        savedProduct.setDescription("Updated Description");
+        savedProduct.setPrice(BigDecimal.valueOf(20.00));
+        savedProduct.setActive(true);
+        Product updatedProduct = productRepository.save(savedProduct);
```

Comment:
tutaj tak samo jak we wcześniejszych testach moje podejrzenie

```suggestion
        productRepository.save(product);
        Product savedProduct = productRepository.findById(product.getId()).get();
                // When
        savedProduct.setName("Updated Product Name");
        savedProduct.setDescription("Updated Description");
        savedProduct.setPrice(BigDecimal.valueOf(20.00));
        savedProduct.setActive(true);
        Product updatedProduct = productRepository.findById(savedProduct.getId()).get(
```